base/channels.jl

```diff
@@ -129,12 +135,13 @@ julia> String(collect(chnl))
 "hello world"
 ```
 """
-function Channel{T}(func::Function, size=0; taskref=nothing, spawn=false) where T
+function Channel{T}(func::Function, size=0; taskref=nothing, spawn=false, threadpool=:default) where T
```

Comment:
Please add a few unit tests.

At least:
- testing that the task is created on the expected threadpool, for both options
- testing the default, if it's not set
- testing what happens if you set threadpool but spawn=false.

Reply:
looks great thanks